linkerd/stack/src/spawn_clone_response.rs

```diff
@@ -0,0 +1,130 @@
+use crate::{layer, NewService, Service, ServiceExt};
+use futures::{future, prelude::*};
+use linkerd_error::Error;
+use std::{
+    fmt,
+    future::Future,
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+};
+use tokio::sync::watch;
+
+/// A `NewService<T>` that produces a [`ThunkCloneResponse`] over an inner
+/// `Service<T>`, calling that service a single time in a background task and
+/// producing `Service<()>`s that clone the response of that call.
+#[derive(Debug, Clone)]
+pub struct SpawnThunkCloneResponse<S> {
```

Comment:
another name alternative could be `SpawnMemoize` (`NewService`) and `Memoized` (`Service`)?